src/main/java/org/opensearch/sdk/ClusterStateResponseHandler.java

```diff
@@ -0,0 +1,40 @@
+package org.opensearch.sdk;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.cluster.ExtensionClusterStateResponse;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+
+public class ClusterStateResponseHandler implements TransportResponseHandler<ExtensionClusterStateResponse> {
+    private static final Logger logger = LogManager.getLogger(ClusterStateResponseHandler.class);
+    final CountDownLatch inProgressLatch = new CountDownLatch(1);
```

Comment:
Created this handler only during the bootstrap just to make sure ClusterState is accessible from OpenSearch. Since SDK will work as a boiler plate and later based on whichever plugin integrates SDK will create their own handler for on their use case.

Reply:
SDK will consist of all the extension points mentioned [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/plugins/Plugin.java). Based on the requirement of the plugin they will access the extension points when integrating with SDK and making their plugin to work as an extension. These extension points will provide a hook to connect the extension with OpenSearch code through Transport Request/Response.

ClusterState, ClusterSettings, LocalNode will be required by createComponent extension point.
More details on the design is [here](https://github.com/opensearch-project/OpenSearch/issues/2447).